Added code to support go version < 1.8. and fixed the issue #3103

pontiyaraja opened the pull request:
Just I copied the same method from "net/url" `Hostname  and Port` to client.go
Its working fine in go version 1.7 to 1.8. Actually I tested it from go 1.7 to 1.8 only.

@evanphx @hornbeck @ctennis @mitchellh @abedra Please review my code

jefferai commented:
Vault requires Go 1.8. it is documented in our readme. When 1.9 is released (it's in RC), Go 1.7 will be unsupported by upstream.

pontiyaraja commented:
Some time Travis CI is failing to build, first I got  go fmt error that is ok, second time it failed to connect with db, then I added just some comments and updated my pull request, finally it was success at my 3rd push, actually I did nothing just added comments in my 3rd pull request. Do you guys review public pull requests even it fails  ???
